### ISSUE TITLE
create-next-app by example when the boilerplate has husky, doens't work 

### ISSUE THREAD

**Issue by renanlimaleite** [vercel/next.js#55342]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #92~20.04.1-Ubuntu SMP Mon Aug 21 14:00:49 UTC 2023
    Binaries:
      Node: 18.17.1
      npm: 9.6.7
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.19
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.2
      husky: "8.0.3",
    Next.js Config:
      output: N/A
```

### Which example does this report relate to?

create-next-app by example

### Describe the Bug

The bug you are encountering is related to the create-next-app command not initializing a git repository, which is causing the husky install command to fail because it can't find the .git directory.

When you run the create-next-app command with the --example flag, it is supposed to create a new Next.js app using the specified example. However, in your case, it seems to be failing during the installation of packages, specifically when trying to run the husky install command, which is part of the prepare script in package.json.

The husky install command requires the project to be a git repository (i.e., it needs a .git directory), but the create-next-app command does not seem to be initializing a git repository. This causes the husky install command to fail, which in turn causes the npm install command to fail, and ultimately the entire create-next-app command fails.

The error message you are seeing, husky - .git can't be found, indicates that the husky install command cannot find the .git directory. This is consistent with the create-next-app command not initializing a git repository.

### Expected Behavior

The expected behavior when you run the create-next-app command with the --example flag is that it should create a new Next.js app using the specified example, install all the necessary packages, and set up the project correctly.

In your specific case, the expected behavior would be:

The create-next-app command downloads the files from the specified GitHub repository.
It creates a new directory for your project and places the downloaded files there.
It installs all the necessary packages, including husky.
The husky install command runs successfully because a .git directory has been created in the project directory.
The entire installation process completes successfully, and you are left with a fully functional Next.js app that you can start developing.
In other words, the create-next-app command should handle initializing the git repository and installing husky and other packages without any errors.

### To Reproduce

Run this command: 
npx create-next-app --example https://github.com/renanlimaleite/boilerplate-next-app-router

**Comment by balazsorban44** [comment]
Hi, this has been supported for a while, added back in 2019, see: https://github.com/vercel/next.js/pull/9376

I tried your reproduction and it was set up correctly.

In case it was successful, you should be seeing the following message in the console:

> Initialized a git repository.
![image](https://github.com/vercel/next.js/assets/18369201/a69215a5-0320-4dca-aea3-7fab0c802238)


There might have been an issue with running `git init` in your case, which won't stop bootstrapping the app. So this works as expected.
